---
layout: post
title:  "ALife OEE3 Paper: Understanding Fraternal Transitions in Individuality"
date:   2018-05-23
---

> This paper provides profound insights for GA design, in specific the automatic generation of evolvable genotype-phenotype mappings using ANN autoencoders.
> It is clearly elucidated, with a firm grasp of the foundational literature in both biological evolution and evolutionary computation.
> The results are promising and establish a rich line of inquiry for researchers to pursue.

--- Reviewer #1

Thanks, Reviewer #1.
My co-authors ([Charles Ofria](http://www.ofria.com/) and [Wolfgang Banzhaf](http://www.cse.msu.edu/~banzhafw/)) and I are really excited about *Learning an Evolvable Genotype-Phenotype Mapping*, too!
(Almost as excited as we are about traveling to Kyoto this summer to present it at the [2018 GECCO conference](http://gecco-2018.sigevo.org/).)
I hope this blog post will get you on board, too, or at least prompt some good chin-scratching (the academic equivalent of exuberance! :smile:).

We will leave out most details about what we actually did in the paper (you can find those in the paper!), and instead focus on describing and discussing the paper's main idea: a new digital evolution technique we call AutoMap.
We'll start by reflecting on the problem this technique tries to solve, introduce the deep learning methods the technique employs (called "autoencoders") then cover how the technique works, briefly review the experiments performed in the paper, and prognosticate on where the research goes from here.
This blog post will focus on building intuition for the what and why of AutoMap, especially for folks outside the field.
We'll use lots of graphics, skip the math, emphasize the biological analogy of digital evolution, and throw in fun bits along the way.

If you're just looking for the publication itself, you can find the PDF [here](https://github.com/mmore500/gecco-2018/releases) and the paper's supporting materials [here](https://osf.io/n92c7/).

## What is an Organism?

![](https://osf.io/cbqht/download){:width="50%"}

[Gerard Morris](https://www.pugetsound.edu/faculty-pages/gmorris/)

[Mallory Thompson](http://www.music.northwestern.edu/faculty/profiles/mallory-thompson.html)

We're all used to thinking of ourselves, as organisms, of course.
But what about our constituent cells? Are they organisms?

tasmanian devil face tumors
https://www.nature.com/articles/439549a


dicty
https://www.youtube.com/watch?v=bkVhLJLG7ug

symbiosis
lichens

everyone's favorite
clownfish and anemone

"What kind of fish is that?"
"It's Nemo!"

biofilms



colonies of multicellular insects are an organism?
e.g., multicellular organisms, eusocial insect colonies

# Why care? Genesis of Complexity

# What did you actually do?

Set of simple rules to do this
"Selection on Kin Cooperating Group Structure Yields Transitions of Individuality"
on a toroidal grid.
(This was a potential title).

Let's get back to this in a second and walk through the simulation setup.

lots of graphics here.
* animation of resource wave

![](https://osf.io/4xfjc/download)

* inheritance of channels


* animation of activation quiescence signaling
![](https://osf.io/g5ye8/download)


* summary graphic

![](https://osf.io/42nwd/download)

![](https://osf.io/r8xfz/download)

All this is described in detail in the paper

# Where to from here?

* experiments with SignalGP
* parallelism!
* increase complexity of individual

## Acknowledgements

Thanks, of course, to my coauthors [Charles Ofria](http://www.ofria.com/) and [Wolfgang Banzhaf](http://www.cse.msu.edu/~banzhafw/) [one of whom](http://www.ofria.com/) also deserves a shout-out for serving as my PhD advisor.

I'd also like to acknowledge Professors [America Chambers](http://mathcs.pugetsound.edu/~alchambers/) and [Adam Smith](http://mathcs.pugetsound.edu/~aasmith/) for advising my initial theoretical and experimental work on evolvability, which led me to this methodological research.

This writeup benefited from feedback from members of [my lab group](http://devolab.msu.edu/). Thank you!

As a graduate student, I receive institutional support from the [Computer Science and Engineering](http://www.cse.msu.edu/) and the [NSF BEACON Center for the Study of Evolution in Action](https://www3.beacon-center.org/) at [Michigan State University](https://msu.edu/).

## References

<a name="Allen2018GenerativeCompletion"
href="https://www.cc.gatech.edu/~hays/7476/projects/Avery_Wenchen/">
Allen and Li.
"Generative Adversarial Denoising Autoencoder for Face Completion."
*CS 7476 Advanced Computer Vision Project @ Georgia Tech College of Computing* (2018).
</a>

<a name="Cheney2013UnshacklingEncoding"
href="https://dl.acm.org/citation.cfm?id=2463404">
Cheney, Nick, et al.
"Unshackling evolution: evolving soft robots with multiple materials and a powerful generative encoding."
*Proceedings of the 15th annual conference on Genetic and evolutionary computation.*
ACM, 2013.
</a>

<a name="Clune2008HowDecreases"
href="https://link.springer.com/chapter/10.1007/978-3-540-87700-4_36">
Clune, Jeff, Charles Ofria, and Robert T. Pennock.
"How a generative encoding fares as problem-regularity decreases."
*International Conference on Parallel Problem Solving from Nature.*
Springer, Berlin, Heidelberg, 2008.
</a>

<a name="Clune2009HybrIDComputation"
href="https://link.springer.com/chapter/10.1007/978-3-642-21314-4_17">
Clune, Jeff, et al.
"HybrID: A hybridization of indirect and direct encodings for evolutionary computation."
*European Conference on Artificial Life.*
Springer, Berlin, Heidelberg, 2009.
</a>

<a name="Coyne1987LackMelanogaster"
href="https://www.ncbi.nlm.nih.gov/pubmed/3108362">
Coyne, J. A.
"Lack of response to selection for direction asymmetry in Drosophila melanogaster."
*Journal of Heredity*, 78(119) (1987).
</a>

<a name="Dertat2017AppliedAutoencoders"
href="https://towardsdatascience.com/applied-deep-learning-part-3-autoencoders-1c083af4d798">
Dertat, Arden.
"Applied Deep Learning - Part 3: Autoencoders."
*Towards Data Science @ Medium* (2017).
</a>

<a name="Despois2017Latent#2"
href="https://hackernoon.com/latent-space-visualization-deep-learning-bits-2-bd09a46920df">
Despois, Julien.
"Latent space visualization — Deep Learning bits #2."
*Hacker Noon @ Medium* (2017).
</a>

<a name="Devesa2016MultipleGrowth"
href="https://www.ncbi.nlm.nih.gov/pubmed/27773998">
Devesa, J., et al.
*Multiple Effects of Growth Hormone in the Body: Is it Really the Hormone for Growth?
*Clinical medicine insights Endocrinology and diabetes*, 9:47–71 (2016).
</a>

<a name="Frowd2004EvoFITComposites"
href="https://dl.acm.org/citation.cfm?id=1008725">
Frowd, Charlie D., et al.
"EvoFIT: A holistic, evolutionary facial imaging technique for creating composites."
*ACM Transactions on applied perception (TAP)* 1.1 (2004): 19-39.
</a>

<a name="Griffiths2015IntroductionAnalysis"
href="https://www.amazon.com/Introduction-Genetic-Analysis-Anthony-Griffiths-ebook/dp/B00SI4UW6C">
Griffiths, A. et al.
"Introduction to genetic analysis."
New York, NY: W.H. Freeman & Company (2015).
</a>

<a name="Hornby2006AutomatedAlgorithms"
href="https://arc.aiaa.org/doi/abs/10.2514/6.2006-7242">
Hornby, Gregory, et al.
"Automated antenna design with evolutionary algorithms."
*Space 2006.* 2006. 7242.
</a>

<a name="Karras2017ProgressiveVariation"
href="https://arxiv.org/abs/1710.10196">
Karras, Tero, et al.
"Progressive growing of gans for improved quality, stability, and variation."
*arXiv preprint*
arXiv:1710.10196 (2017).
</a>

<a name="Kouvaris2017HowOrganisation"
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5383015/">
Kouvaris, Kostas, et al.
"How evolution learns to generalise: Using the principles of learning theory to understand the evolution of developmental organisation."
*PLoS computational biology* 13.4 (2017): e1005358.
</a>

<a name="Rimbault2013DerivedBreeds"
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3847769/">
Rimbault, M. et al.
"Derived variants at six genes explain nearly half of size reduction in dog breeds."
*Genome research*, 23(12):1985–95 (2013).
</a>

<a name="Tuinstra1990LackDevelopment"
href="http://rspb.royalsocietypublishing.org/content/241/1301/146">
Tuinstra, E., et al.
"Lack of response to family selection for direction asymmetry in Drosophila melanogaster: left and right are not distinguished during development."
Proc. R. Soc. Lond. B, 241(1301):146–152 (1990).
</a>

<a name="White2016SamplingTechniques"
href="https://arxiv.org/abs/1609.04468">
White, Tom.
"Sampling generative networks: Notes on a few effective techniques."
*arXiv preprint*
arXiv:1609.04468 (2016).
</a>
